label_studio/core/middleware.py

```diff
@@ -153,3 +154,21 @@ def process_view(self, request, view_func, view_args, view_kwargs):
         if hasattr(request, 'user') and request.method not in SAFE_METHODS:
             if request.user.is_authenticated:
                 request.user.update_last_activity()
+
+class InactivitySessionTimeoutMiddleWare(CommonMiddleware):
+    """Log the user out if they have been logged in for too long
+     or inactive for too long"""
+
+    def process_request(self, request) -> None:
+        if not hasattr(request, "session") or request.session.is_empty():
+            return
+
+        current_time = time.time()
+        last_login = request.session['last_login'] if 'last_login' in request.session else 0
+
+        # Check if this request is too far from when the login happened
+        if (current_time - last_login) > settings.MAX_SESSION_AGE:
```

Comment:
In what case is last_login not in request.session, and therefore set to 0? In that case we will log the user out here right?

Reply:
Correct. The effect here is that everyone will be logged out whenever this update is pushed since the session doesn't know how old the login is. Maybe that's too aggressive? But it will only happen once ever. I opted not to try to write a "now" default since that would be slightly more complex and could possibly open the session up to be extended indefinitely.